src/client/auth/auth.go

```diff
@@ -213,3 +231,25 @@ func IsErrTooShortTTL(err error) bool {
 		strings.Contains(errMsg, ") is shorter than token's existing TTL (") &&
 		strings.Contains(errMsg, ")")
 }
+
+// HashToken converts a token to a cryptographic hash.
+// We don't want to store tokens verbatim in the database, as then whoever
+// that has access to the database has access to all tokens.
+func HashToken(token string) string {
+	sum := sha256.Sum256([]byte(token))
+	return fmt.Sprintf("%x", sum)
+}
+
+// GetAuthToken extracts the auth token embedded in 'ctx', if there is one
+func GetAuthToken(ctx context.Context) (string, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return "", ErrNoMetadata
+	}
+	if len(md[ContextTokenKey]) > 1 {
+		return "", errors.Errorf("multiple authentication token keys found in context")
+	} else if len(md[ContextTokenKey]) == 0 {
+		return "", ErrNotSignedIn
+	}
+	return md[ContextTokenKey][0], nil
+}
```

Comment:
Refactor these out of the auth server into the auth client package. They're useful in tests as well as the admin logic.